ui/pages/confirm-token-transaction-base/confirm-token-transaction-base.js

```diff
@@ -10,26 +11,57 @@ import {
   addFiat,
   roundExponential,
 } from '../../helpers/utils/confirm-tx.util';
-import { getWeiHexFromDecimalValue } from '../../helpers/utils/conversions.util';
-import { ETH, PRIMARY } from '../../helpers/constants/common';
+import {
+  getWeiHexFromDecimalValue,
+  hexWEIToDecETH,
+} from '../../helpers/utils/conversions.util';
+import {
+  ERC1155,
+  ERC20,
+  ERC721,
+  ETH,
+  PRIMARY,
+} from '../../helpers/constants/common';
+import {
+  contractExchangeRateSelector,
+  getCurrentCurrency,
+} from '../../selectors';
+import {
+  getConversionRate,
+  getNativeCurrency,
+} from '../../ducks/metamask/metamask';
 
 export default function ConfirmTokenTransactionBase({
-  image,
-  title,
-  subtitle,
+  image = '',
+  assetName,
   toAddress,
   tokenAddress,
   tokenAmount = '0',
-  fiatTransactionTotal,
-  ethTransactionTotal,
-  ethTransactionTotalMaxAmount,
-  contractExchangeRate,
-  conversionRate,
-  currentCurrency,
-  nativeCurrency,
+  tokenSymbol,
+  tokenId,
+  assetStandard,
   onEdit,
+  ethTransactionTotal,
+  fiatTransactionTotal,
+  hexMaximumTransactionFee,
 }) {
   const t = useContext(I18nContext);
+  const contractExchangeRate = useSelector(contractExchangeRateSelector);
+  const nativeCurrency = useSelector(getNativeCurrency);
+  const currentCurrency = useSelector(getCurrentCurrency);
+  const conversionRate = useSelector(getConversionRate);
+
+  const ethTransactionTotalMaxAmount = Number(
```

Comment:
`toFixed` will add extra zeroes if the number being passed is less than the precision given. e.g. `Number(3.2).toFixed(6)` => `3.200000`. Is that okay?

Reply:
Confused now about why I added this... I don't think we want to limit precision here. Will remove.

Reply:
fixed here: https://github.com/MetaMask/metamask-extension/pull/13788/commits/6fb5f7e725635b5f364fe4ec7d0d13024284b59a